Code/Framework/AzFramework/AzFramework/Physics/Collision/CollisionGroups.h

```diff
@@ -174,4 +173,9 @@ namespace AzPhysics
     private:
         AZStd::vector<Preset> m_groups;
     };
+
+    //! Construct a Group with the given Id of a collision group.
+    //! This will lookup the group Id to retrieve the group mask. If not found, CollisionGroup::All is returned.
+    //! @param id The Id of the group to look up the group mask.
+    CollisionGroup MakeCollisionGroup(const CollisionGroups::Id& id);
```

Comment:
question: Is ```CollisionGroup::All``` a good default to return?
i wonder if we used a ```AZStd::optional``` as the return so the caller could determine if the group was not found and handle that as needed.

Reply:
The default return value is actually dictated by the existing functionality in `CollisionGroups::FindGroupById(CollisionGroups::Id id).`

This helper function is just a wrapper around `Physics::CollisionRequestBus::BroadcastResult(group, &Physics::CollisionRequests::GetCollisionGroupById, id);` to compensate for the missing constructor `CollisionGroup(const CollisionGroups::Id& id);` which we can't have due to `class Id` being an inner one that can't be forward-declared.

The other constructor does the same, so being consistent here:
 
```
//! Construct a Group with the given name.
//! This will lookup the group name to retrieve the group mask. If not found, CollisionGroup::All is set.
//! @param groupName The name of the group to look up the group mask.
CollisionGroup(const AZStd::string& groupName);
```

Reply:
From the description and implementation it seems like `GetCollisionGroup` or `RetrieveCollisionGroup` would be more accurate names.

Reply:
Yeah, made it the same name as the ebus function now